src/main.rs

```diff
@@ -65,12 +64,27 @@ fn main() -> anyhow::Result<()> {
     let otlp_conf = psh_config.otlp_conf();
     let mut otlp_th = None;
 
+    let runtime = Arc::new(tokio::runtime::Runtime::new()?);
+
+    let luoxiao = psh_config.luoxiao();
+    let addr = luoxiao.addr().to_owned();
+    let token = luoxiao.token().to_owned();
+    runtime.spawn(async move {
+        match services::rpc::Luoxiao::new(&addr).await {
+            Ok(mut lx) => {
+                if let Err(e) = lx.send_info(token).await {
+                    tracing::error!("send info to luoxiao: {}", e)
+                };
+            }
+            Err(e) => tracing::error!("connect to luoxiao: {}", e),
+        }
+    });
+
     if otlp_conf.enable() {
+        let rt = Arc::clone(&runtime);
         let export_conf: ExportConfig = otlp_conf.into();
-        let th = std::thread::spawn(|| {
-            tokio::runtime::Runtime::new()?.block_on(otlp::otlp_tasks(export_conf))
-        });
-        otlp_th = Some(th);
+        let hd = std::thread::spawn(move || rt.block_on(otlp::otlp_tasks(export_conf)));
```

Comment:
Have you tested this branch?  I doubt it wouldn't work as `rt` is not initialized before the new created thread `hd`,  my understanding is that tokio runtime manages a bunch of threads, and set thread local variable to reference to tokio runtime, a user created thread may not be managed by tokio and thus cannot simply "move" a tokio runtime into it.

Reply:
It works fine. In the tokio doc [share runtime](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#sharing) there is a way to share runtime and run on another thread and prevent the runtime from shutting down.

Reply:
Well, it is true you can share a tokio runtime between threads by using `Arc`. The situation becomes if it is necessary to share this tokio runtime with a new created thread.  Will `otlp::otlp_tasks(export_confg)` run inside this new created thread or it is managed by tokio runtime, no matter this new thread? 

You can do an experiment, comment `PshEngineBuilder` code to see if psh will exit immediately, if so, then create a new thread here is not necessary.

Reply:
Psh not exit immediately.